astropy/visualization/wcsaxes/coordinate_helpers.py

```diff
@@ -861,3 +896,123 @@ def _update_grid_contour(self):
+        :meth:`~astropy.visualization.wcsaxes.CoordinateHelper.set_ticklabel`,
+        :meth:`~astropy.visualization.wcsaxes.CoordinateHelper.set_ticks_position`,
+        :meth:`~astropy.visualization.wcsaxes.CoordinateHelper.set_ticklabel_position`,
+        and :meth:`~astropy.visualization.wcsaxes.CoordinateHelper.grid`.
+
+        Parameters
+        ----------
+        which : {'major', 'minor', 'both'}, optional
+            Which ticks to apply the settings to. By default, setting are
+            applied to both major and minor ticks. Note that if ``'minor'`` is
+            specified, only the length of the ticks can be set currently.
+        direction : {'in', 'out'}, optional
+            Puts ticks inside the axes, or outside the axes.
+        length : float, optional
+            Tick length in points.
+        width : float
+            Tick width in points.
+        color : color, optional
+            Tick color (accepts any valid Matplotlib color)
+        pad : float, optional
+            Distance in points between tick and label.
+        labelsize : float or str, optional
+            Tick label font size in points or as a string (e.g., 'large').
+        labelcolor : color, optional
+            Tick label color (accepts any valid Matplotlib color)
+        colors : color, optional
+            Changes the tick color and the label color to the same value
+             (accepts any valid Matplotlib color).
+        bottom, top, left, right : bool, optional
+            Where to draw the ticks. Note that this will not work correctly if
+            the frame is not rectangular.
+        labelbottom, labeltop, labelleft, labelright : bool, optional
+            Where to draw the tick labels. Note that this will not work
+            correctly if the frame is not rectangular.
+        grid_color : color, optional
+            The color of the grid lines (accepts any valid Matplotlib color).
+        grid_alpha : float, optional
+            Transparency of grid lines: 0 (transparent) to 1 (opaque).
+        grid_linewidth : float, optional
+            Width of grid lines in points.
+        grid_linestyle : string, optional
+            The style of the grid lines (accepts any valid Matplotlib line
+            style).
+        """
+
+        # First do some sanity checking on the keyword arguments
+
+        # colors= is a fallback default for color and labelcolor
+        if 'colors' in kwargs:
+            if 'color' not in kwargs:
+                kwargs['color'] = kwargs['colors']
+            if 'labelcolor' not in kwargs:
+                kwargs['labelcolor'] = kwargs['colors']
+
+        # The only property that can be set *specifically* for minor ticks is
+        # the length. In future we could consider having a separate Ticks instance
+        # for minor ticks so that e.g. the color can be set separately.
+        if which == 'minor':
+            if set(kwargs) - {'length'}:
```

Comment:
This works (at least now), but I generally try to use explicit boolean statements, e.g. `len(set(kwargs) - {'length'}) > 0`.  How robust is it for an non-empty set to always evaluate to `True` (and will that ever change)?

Reply:
I agree, fixed